src/mpi/coll/ibcast/ibcast.c

```diff
@@ -232,77 +322,98 @@ int MPIR_Ibcast_impl(void *buffer, int count, MPI_Datatype datatype, int root,
-                mpi_errno =
-                    MPIR_Ibcast_intra_gentran_tree(buffer, count, datatype, root, comm_ptr,
-                                                   MPIR_Ibcast_tree_type,
-                                                   MPIR_CVAR_IBCAST_TREE_KVAL,
-                                                   MPIR_CVAR_IBCAST_TREE_PIPELINE_CHUNK_SIZE,
-                                                   request);
-                break;
-
-            case MPIR_CVAR_IBCAST_INTRA_ALGORITHM_gentran_scatterv_recexch_allgatherv:
-                mpi_errno =
-                    MPIR_Ibcast_intra_gentran_scatterv_recexch_allgatherv(buffer, count, datatype,
-                                                                          root, comm_ptr,
-                                                                          MPIR_CVAR_IBCAST_SCATTERV_KVAL,
-                                                                          MPIR_CVAR_IBCAST_ALLGATHERV_RECEXCH_KVAL,
-                                                                          request);
-                break;
-
-            case MPIR_CVAR_IBCAST_INTRA_ALGORITHM_gentran_ring:
-                mpi_errno =
-                    MPIR_Ibcast_intra_gentran_ring(buffer, count, datatype, root, comm_ptr,
-                                                   MPIR_CVAR_IBCAST_RING_CHUNK_SIZE, request);
-                break;
-
-            case MPIR_CVAR_IBCAST_INTRA_ALGORITHM_sched_binomial:
-                MPII_SCHED_WRAPPER(MPIR_Ibcast_intra_sched_binomial, comm_ptr, request, buffer,
-                                   count, datatype, root);
-                break;
-
-            case MPIR_CVAR_IBCAST_INTRA_ALGORITHM_sched_smp:
-                MPII_SCHED_WRAPPER(MPIR_Ibcast_intra_sched_smp, comm_ptr, request, buffer,
-                                   count, datatype, root);
-                break;
-
-            case MPIR_CVAR_IBCAST_INTRA_ALGORITHM_sched_scatter_recursive_doubling_allgather:
-                MPII_SCHED_WRAPPER(MPIR_Ibcast_intra_sched_scatter_recursive_doubling_allgather,
-                                   comm_ptr, request, buffer, count, datatype, root);
-                break;
-
-            case MPIR_CVAR_IBCAST_INTRA_ALGORITHM_sched_scatter_ring_allgather:
-                MPII_SCHED_WRAPPER(MPIR_Ibcast_intra_sched_scatter_ring_allgather, comm_ptr,
-                                   request, buffer, count, datatype, root);
-                break;
-
-            case MPIR_CVAR_IBCAST_INTRA_ALGORITHM_sched_auto:
-                MPL_FALLTHROUGH;
-
-            default:
-                MPII_SCHED_WRAPPER(MPIR_Ibcast_intra_sched_auto, comm_ptr, request, buffer, count,
-                                   datatype, root);
-                break;
+        int i;
+        for (i = 0; MPIR_CVAR_IBCAST_INTRA_ALGORITHM[i] != MPIR_CVAR_ENUM_NULL; i++) {
+            switch (MPIR_CVAR_IBCAST_INTRA_ALGORITHM[i]) {
```

Comment:
This PR does not include OFI collective selection, so it is not a concern. But what I said was that if this code is applied to OFI level selection, we need to have a switch statement to include all algorithms including the ones in MPIR level. So the users can say MPIR_CVAR_IBCAST_INTRA_ALGORITHM=ofi_alg1,mpir_alg2.

Reply:
No, that should never be the case.  For a fully tuned library, it should be: `MPIR_CVAR_OFI_IBCAST_INTRA_ALGORITHM=ofi_alg1,ofi_alg2,mpir_ibcast`